packages/@aws-cdk/aws-apigateway/test/authorizers/test.lambda.ts

```diff
@@ -76,13 +112,46 @@ export = {
     test.done();
   },
 
+  'request authorizer with all parameters specified'(test: Test) {
+    const stack = new Stack();
+
+    const func = new lambda.Function(stack, 'myfunction', {
+      handler: 'handler',
+      code: lambda.Code.fromInline('foo'),
+      runtime: lambda.Runtime.NODEJS_12_X,
+    });
+
+    const auth = new RequestAuthorizer(stack, 'myauthorizer', {
+      handler: func,
+      identitySource: 'method.request.header.whoami',
+      authorizerName: 'myauthorizer',
+      resultsCacheTtl: Duration.minutes(1),
+    });
+
+    const restApi = new RestApi(stack, 'myrestapi');
+    restApi.root.addMethod('ANY', undefined, {
+      authorizer: auth,
+      authorizationType: AuthorizationType.CUSTOM
+    });
+
+    expect(stack).to(haveResource('AWS::ApiGateway::Authorizer', {
+      Type: 'REQUEST',
+      RestApiId: stack.resolve(restApi.restApiId),
+      IdentitySource: 'method.request.header.whoami',
+      Name: 'myauthorizer',
+      AuthorizerResultTtlInSeconds: 60
+    }));
+
+    test.done();
+  },
+
   'token authorizer with assume role'(test: Test) {
     const stack = new Stack();
 
     const func = new lambda.Function(stack, 'myfunction', {
       handler: 'handler',
       code: lambda.Code.fromInline('foo'),
-      runtime: lambda.Runtime.NODEJS_8_10,
+      runtime: lambda.Runtime.NODEJS_12_X,
```

Comment:
We don't yet support inline code for `NODEJS_12_X` unless this is merged in before yours - https://github.com/aws/aws-cdk/pull/5710. If not, use 10_X